aspnetcore/fundamentals/static-files.md

```diff
@@ -4,68 +4,281 @@ author: rick-anderson
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupRose.cs?name=snippet&highlight=14-21)]
 
-::: moniker-end
+In the preceding code, the *MyStaticFiles* directory hierarchy is exposed publicly via the *StaticFiles* URI segment. A request to `https://<hostname>/StaticFiles/images/red-rose.jpg` serves the *red-rose.jpg* file.
+
+The following markup references *MyStaticFiles/images/red-rose.jpg*:
+
+```html
+<img src="~/StaticFiles/images/red-rose.jpg" class="img" alt="A red rose." />
+```
+
+### Set HTTP response headers
+
+A <xref:Microsoft.AspNetCore.Builder.StaticFileOptions> object can be used to set HTTP response headers. In addition to configuring static file serving from the [web root](xref:fundamentals/index#web-root), the following code sets the `Cache-Control` header:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupAddHeader.cs?name=snippet&highlight=14-23)]
+
+Static files are publicly cacheable for 600 seconds in the Development environment:
+
+![Response headers showing the Cache-Control header has been added](static-files/_static/add-header.png)
+
+## Static file authorization
+
+The Static File Middleware doesn't provide authorization checks. Any files served by it, including those under `wwwroot`, are publicly accessible. To serve files based on authorization:
+
+* Store them outside of `wwwroot` and any directory accessible to the Static File Middleware.
+* Serve them via an action method to which authorization is applied. Return a [FileResult](/dotnet/api/microsoft.aspnetcore.mvc.fileresult) object:
+
+  [!code-csharp[](static-files/samples/1x/Controllers/HomeController.cs?name=snippet_BannerImageAction)]
+
+## Directory browsing
+
+Directory browsing allows directory listing and files within specified directories. Directory browsing is disabled by default for security reasons, see [Considerations](#sc). Enable directory browsing with:
+
+* <xref:Microsoft.Extensions.DependencyInjection.DirectoryBrowserServiceExtensions.AddDirectoryBrowser%2A> in `Startup.ConfigureServices`.
+* [UseDirectoryBrowser](/dotnet/api/microsoft.aspnetcore.builder.directorybrowserextensions.usedirectorybrowser#Microsoft_AspNetCore_Builder_DirectoryBrowserExtensions_UseDirectoryBrowser_Microsoft_AspNetCore_Builder_IApplicationBuilder_Microsoft_AspNetCore_Builder_DirectoryBrowserOptions_) in `Startup.Configure`.
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupBrowse.cs?name=snippet&highlight=4,20-34)]
+
+The preceding code allows directory browsing of the *wwwroot/images* folder using the URL `https://<hostname>/MyImages`, with links to each file and folder:
+
+![directory browsing](static-files/_static/dir-browse.png)
+
+See [Security considerations](#sc) on the security risks when enabling browsing.
+
```

Comment:
```suggestion
```

I think this can be dropped, because it's called out before the code example, and my previous suggestion makes that a bit more prominent, too.